From f1b598f0222ce0d580b67ea5a2833af068493653 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:31:22 -0400
Subject: [iommu] add IOMMU_EXEC flag for safely allowing XN mappings

Message-id: <20140905013122.30867.61272.stgit@gir.stowe>
Patchwork-id: 91266
O-Subject: [RHEL7.1 PATCH 02/28] iommu: add IOMMU_EXEC flag for safely allowing XN mappings
Bugzilla: 1129808
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129808
Upstream Status: ca13bb3d4e54f5c30089e55d5b8f4bb80c8f05e2

commit ca13bb3d4e54f5c30089e55d5b8f4bb80c8f05e2
Author: Will Deacon <will.deacon@arm.com>
Date:   Tue Nov 5 15:59:53 2013 +0000

    iommu: add IOMMU_EXEC flag for safely allowing XN mappings

    Whilst most IOMMU mappings should probably be non-executable, there
    may be cases (HSA?) where executable mappings are required.

    This patch introduces a new mapping flag, IOMMU_EXEC, to indicate that
    the mapping should be mapped as executable.

    Signed-off-by: Will Deacon <will.deacon@arm.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index c5fe497..39e02a3 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -24,9 +24,10 @@
 #include <linux/types.h>
 #include <trace/events/iommu.h>
 
-#define IOMMU_READ (1)
-#define IOMMU_WRITE (2)
-#define IOMMU_CACHE (4) /* DMA cache coherency */
+#define IOMMU_READ (1 << 0)
+#define IOMMU_WRITE (1 << 1)
+#define IOMMU_CACHE (1 << 2) /* DMA cache coherency */
+#define IOMMU_EXEC (1 << 3)
 
 struct iommu_ops;
 struct iommu_group;
-- 
1.7.1